src/test/regress/expected/function_propagation.out

```diff
@@ -316,6 +316,616 @@ BEGIN
+                           pg_identify_object_as_address
+---------------------------------------------------------------------
+ (function,"{function_propagation_schema,func_in_transaction_for_local_table}",{})
+(1 row)
+
+ROLLBACK;
+BEGIN;
+    CREATE OR REPLACE FUNCTION exclude_bool_func()
+    RETURNS boolean
+    LANGUAGE plpgsql IMMUTABLE AS
+    $$
+    BEGIN
+        return true;
+    END;
+    $$;
+    -- Functions shouldn't be propagated within transaction
+    SELECT pg_identify_object_as_address(classid, objid, objsubid) from citus.pg_dist_object where objid = 'function_propagation_schema.exclude_bool_func'::regproc::oid;
+ pg_identify_object_as_address
+---------------------------------------------------------------------
+(0 rows)
+
+    CREATE TABLE exclusion_func_prop_table (id int, EXCLUDE USING btree (id WITH =) WHERE (exclude_bool_func()));
+    SELECT create_distributed_table('exclusion_func_prop_table', 'id');
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+    -- Function should be marked as distributed after distributing the table that depends on it
+    SELECT pg_identify_object_as_address(classid, objid, objsubid) from citus.pg_dist_object where objid = 'function_propagation_schema.exclude_bool_func'::regproc::oid;
+                  pg_identify_object_as_address
+---------------------------------------------------------------------
+ (function,"{function_propagation_schema,exclude_bool_func}",{})
+(1 row)
+
+COMMIT;
+-- Function should be marked as distributed on the worker after committing changes
+SELECT * FROM run_command_on_workers($$SELECT pg_identify_object_as_address(classid, objid, objsubid) from citus.pg_dist_object where objid = 'function_propagation_schema.exclude_bool_func'::regproc::oid;$$) ORDER BY 1,2;
+ nodename  | nodeport | success |                             result
+---------------------------------------------------------------------
+ localhost |    57637 | t       | (function,"{function_propagation_schema,exclude_bool_func}",{})
+ localhost |    57638 | t       | (function,"{function_propagation_schema,exclude_bool_func}",{})
+(2 rows)
+
+-- Show that having a function dependency for index also works
+BEGIN;
+    CREATE OR REPLACE FUNCTION func_for_index_predicate(col_1 int)
+    RETURNS boolean
+    LANGUAGE plpgsql IMMUTABLE AS
+    $$
+    BEGIN
+        return col_1 > 5;
+    END;
+    $$;
+    -- Functions shouldn't be propagated within transaction
+    SELECT pg_identify_object_as_address(classid, objid, objsubid) from citus.pg_dist_object where objid = 'function_propagation_schema.func_for_index_predicate'::regproc::oid;
+ pg_identify_object_as_address
+---------------------------------------------------------------------
+(0 rows)
+
+    CREATE TABLE table_to_check_func_index_dep (id int, col_2 int);
```

Comment:
Mostly sharing for reference. 

As far as I can tell, this test already works on the master branch via the text search configuration changes.

And, note that the dependency type is Normal.
```SQL
 SELECT pg_identify_object_as_address(classid, objid, objsubid),deptype from pg_depend where refobjid = 'func_for_index_predicate'::regproc;
┌───────────────────────────────────────────────────────────────┬─────────┐
│                 pg_identify_object_as_address                 │ deptype │
├───────────────────────────────────────────────────────────────┼─────────┤
│ (index,"{public,table_to_check_func_index_dep_col_2_idx}",{}) │ n       │
└───────────────────────────────────────────────────────────────┴─────────┘
(1 row)
```